src/main/java/com/mystays/authorizationserver/handler/ApiExceptionHandler.java

```diff
@@ -0,0 +1,40 @@
+package com.mystays.authorizationserver.handler;
+
+import com.mystays.authorizationserver.model.AuthenticationResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.oauth2.server.resource.InvalidBearerTokenException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class ApiExceptionHandler {
+
+    @ExceptionHandler(BadCredentialsException.class)
+    public ResponseEntity<AuthenticationResponse> handleBadCredentials(BadCredentialsException badCredentialsException){
+        AuthenticationResponse authenticationResponse = new AuthenticationResponse();
+        authenticationResponse.setStatus("Not Authenticated");
+        authenticationResponse.setErrorCode("401");
+        authenticationResponse.setErrorText(badCredentialsException.getMessage());
+        return new ResponseEntity<>(authenticationResponse, HttpStatus.UNAUTHORIZED );
+    }
+
+    @ExceptionHandler(UsernameNotFoundException.class)
+    public ResponseEntity<AuthenticationResponse> handleUserNotFound(UsernameNotFoundException usernameNotFoundException){
+        AuthenticationResponse authenticationResponse = new AuthenticationResponse();
+        authenticationResponse.setStatus("Not Authenticated");
+        authenticationResponse.setErrorCode("401");
+        authenticationResponse.setErrorText(usernameNotFoundException.getMessage());
+        return new ResponseEntity<>(authenticationResponse, HttpStatus.UNAUTHORIZED );
+    }
+
+    @ExceptionHandler(InvalidBearerTokenException.class)
+    public ResponseEntity<AuthenticationResponse> handleInvalidBearerToken(InvalidBearerTokenException invalidBearerTokenException){
+        AuthenticationResponse authenticationResponse = new AuthenticationResponse();
+        authenticationResponse.setErrorCode("401");
+        authenticationResponse.setErrorText(invalidBearerTokenException.getMessage());
+        return new ResponseEntity<>(authenticationResponse, HttpStatus.UNAUTHORIZED );
+    }
+}
```

Comment:
We can add MethodArgumentNotValidException too to handle validation constraints.
`@ExceptionHandler(MethodArgumentNotValidException.class)
    public ResponseEntity<AuthenticationResponse> methodArgumentNotValidExceptionHandler(MethodArgumentNotValidException methodArgumentNotValidException) {
        List<String> errorMessage = new ArrayList<>();
        methodArgumentNotValidException.getBindingResult().getAllErrors().forEach(error -> {
            String fieldName = ((FieldError) error).getField();
            String message = error.getDefaultMessage();
            errorMessage.add(fieldName + " : " + message);
        });
        return ResponseEntityBuilder.build(HttpStatus.BAD_REQUEST, String.join(", ", errorMessage));
    }`